core/src/main/java/tech/pegasys/eth2signer/core/Runner.java

```diff
@@ -99,27 +112,22 @@ public void run() {
     }
   }
 
-  private SigningRequestHandler createSigningHandler(
+  private DirectoryBackedArtifactSignerProvider createSignerProvider(
       final MetricsSystem metricsSystem, final Vertx vertx) {
     final ArtifactSignerFactory artifactSignerFactory =
         new ArtifactSignerFactory(
             config.getKeyConfigPath(), metricsSystem, new HashicorpConnectionFactory(vertx));
-    final DirectoryBackedArtifactSignerProvider signerProvider =
-        new DirectoryBackedArtifactSignerProvider(
-            config.getKeyConfigPath(),
-            "yaml",
-            new YamlSignerParser(artifactSignerFactory),
-            config.getKeyCacheLimit());
-
-    final SigningRequestHandler signingHandler =
-        new SigningRequestHandler(signerProvider, createJsonDecoder());
-    signerProvider.cacheAllSigners();
-
-    return signingHandler;
+    return new DirectoryBackedArtifactSignerProvider(
+        config.getKeyConfigPath(),
+        "yaml",
+        new YamlSignerParser(artifactSignerFactory),
+        config.getKeyCacheLimit());
   }
 
   private Handler<HttpServerRequest> createRouter(
-      final Vertx vertx, final SigningRequestHandler signingHandler) {
+      final Vertx vertx,
```

Comment:
suspect this is now not such a great function, seeing as every new route requires new parameters to this function - should it be refactoredd so that the router gets passed into each handler, which the registers itself? (or similar?)

Reply:
done